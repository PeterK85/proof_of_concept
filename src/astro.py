def test_func(x):
    return x + 85


if __name__ == "__main__":
    print(test_func(0))
    print(test_func(0))
    print(test_func(0))
