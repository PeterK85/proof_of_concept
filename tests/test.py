import sys
sys.path.append('../src')
import astro
import unittest

class Tests(unittest.TestCase):
    def test_1(self):
        self.assertEqual(2+3, 5)
    def test_2(self):
        self.assertEqual(astro.test_func(0), 85)
    def test_fail(self):
        self.assertEqual(astro.test_func(0), 0)


if __name__ == "__main__":
    unittest.main()
