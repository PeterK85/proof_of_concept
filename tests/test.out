..F
======================================================================
FAIL: test_fail (__main__.Tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/peter/dev/proof_of_concept/tests/test.py", line 12, in test_fail
    self.assertEqual(astro.test_func(0), 0)
AssertionError: 85 != 0

----------------------------------------------------------------------
Ran 3 tests in 0.000s

FAILED (failures=1)
